### ISSUE TITLE
Parser gap: an arithmetic range endpoint is read as a subscript flags group

### ISSUE THREAD

**Issue by ss-o** [z-shell/zsh-lint#359]
## Summary

`${a[1,(2+3)]}` is a range whose second endpoint is an arithmetic expression. The front end reads the `(2+3)` as a subscript flags group, producing a `FlagsArithm` whose `Flags.Value` is `2+3`.

The source parses, so nothing fails and no diagnostic is emitted — the tree is simply wrong about what the subscript means.

## Reproduction

```zsh
a=(one two three four five six)
print -r -- "${a[1,(2+3)]}"
```

Native Zsh prints `one two three four five`: the endpoint evaluates to 5 and the range selects elements 1 through 5.

The parser instead produces:

```
ParamExp.Index -> FlagsArithm{Flags: "2+3"}
```

Verified on `main` at `046b7a8` and on the #250 branch; both agree, and `zsh -f -n` accepts the source.

## Mechanism

`flagPatternStart` reads flag letters from the byte after `(` and accepts the group when it reaches `)`. `isFlagLetter` admits any ASCII letter, and the digits and `+` here are not letters — but the scan that reaches this point has already decided the `(` opens a flags group based on position alone.

Subscript flags are a fixed set (`zshparam`, Subscript Flags): `w`, `s`, `n`, `b`, `e`, `i`, `I`, `r`, `R`, `k`, `K`, `p`, `f`, `F`. An arithmetic expression is distinguishable from a flags group by content, and `2+3` holds bytes no flags group can contain.

## Why it is not urgent

A flagged pattern and an arithmetic endpoint are both valid in that position, so the parse succeeds either way and nothing downstream currently reads `FlagsArithm.Flags` in a way that changes a diagnostic. This is a correctness defect in the tree, not a false positive.

It matters for any rule that will reason about subscript flags, and for a consumer reading the AST.

## Acceptance criteria

- `${a[1,(2+3)]}` yields an arithmetic index, not a `FlagsArithm`.
- `${a[1,(i)x]}` keeps its `FlagsArithm` with flags `i`.
- A group mixing letters and operators, `${a[1,(i+1)]}`, is decided one way deliberately and the choice is recorded in a test.
- `TestFlaggedSubscriptArithmeticRangeEndpointIsMisread` in `internal/parse/flagged_subscript_bracket_test.go` is inverted as part of the fix; it currently pins the wrong reading so this defect stays visible.

## Environment

- zsh-lint `main` `046b7a8`
- zsh 5.9.2 (x86_64-pc-linux-gnu)
- mvdan/sh v3.14.1, LangZsh

## References

- #250 / PR #358: found while widening the flagged-pattern opener scan; that change cannot affect this, since the scan runs only after a parse error and this source parses
- #237: the flagged-pattern adapter
